File: api/src/main/java/net/blueshell/api/repository/SponsorRepository.java
package net.blueshell.api.repository;

import net.blueshell.api.base.BaseRepository;
import net.blueshell.api.model.File;
import net.blueshell.api.model.Sponsor;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface SponsorRepository extends BaseRepository<Sponsor, Long> {
    Optional<Sponsor> findByPicture(File picture);
}

File: api/src/main/kotlin/net/blueshell/api/repository/SponsorRepository.kt
package net.blueshell.api.repository

import net.blueshell.api.model.File
import net.blueshell.api.model.Sponsor
import net.blueshell.api.repository.base.BaseRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
@Suppress("FunctionName")
interface SponsorRepository : BaseRepository<Sponsor, Long> {
    fun findBy_picture(picture: File): Optional<Sponsor>
}
